src/hashstore/hashstore.py

```diff
@@ -10,44 +10,133 @@
+    that implement the 'hashstore_interface' abstract methods)
+
+    This factory class provides a method to retrieve a `hashstore` object
+    based on the specified store type (ex. FileHashStore). It supports the
+    creation of different types of hash stores by mapping store types to
+    specific implementations.
+    """
+
+    def __init__(self):
+        """Initialize the HashStoreFactory with default config values"""
+        self.dir_depth = DIR_DEPTH
+        self.dir_width = DIR_WIDTH
+        self.sysmeta_ns = SYSMETA_NS
+        self.default_algo_list = DEFAULT_ALGO_LIST
+        self.other_algo_list = OTHER_ALGO_LIST
+
+    def get_hashstore(self, store_path, hashstore_type):
+        """Get a `HashStore`-like object based on the specified store type.
+
+        Args:
+            root (str): The root directory for the hash store.
+            hashstore_type (str): The type of `HashStore` to retrieve.
+
+        Returns:
+            HashStore: A hash store object based on the given store type.
+
+        Raises:
+            ValueError: If the given store_type is not supported.
+        """
+        hashstore_type.lower()
+        if hashstore_type == "filehashstore":
+            return FileHashStore(
+                root=store_path,
+                depth=self.dir_depth,
+                width=self.dir_width,
+                sysmeta_ns=self.sysmeta_ns,
+                default_algo_list=self.default_algo_list,
+                other_algo_list=self.other_algo_list,
+            )
+        else:
+            raise ValueError(f"hashstore_type: {hashstore_type} is not supported.")
 
 
 class HashStore:
     """HashStore is a content-addressable file management system that
     utilizes a persistent identifier (PID) in the form of a hex digest
     value to address files."""
 
-    # Class variables
-    dir_depth = 3  # The number of directory levels for storing files
-    dir_width = 2  # The width of the directory names, in characters
-    sysmeta_ns = "http://ns.dataone.org/service/types/v2.0"
-    time_out_sec = 1
-    object_lock = threading.Lock()
-    sysmeta_lock = threading.Lock()
-    object_locked_pids = []
-    sysmeta_locked_pids = []
-
     @staticmethod
     def version():
         """Return the version number"""
         __version__ = importlib.metadata.version("hashstore")
         return __version__
 
-    def __init__(self, store_path):
-        """Initialize the hashstore"""
-        self.store_path = store_path
-        self.objects = FileHashStore(
-            self.store_path + "/objects",
-            depth=self.dir_depth,
-            width=self.dir_width,
-            algorithm="sha256",
-        )
-        self.sysmeta = FileHashStore(
-            self.store_path + "/sysmeta",
-            depth=self.dir_depth,
-            width=self.dir_width,
-            algorithm="sha256",
-        )
+    hashstore_factory = HashStoreFactory()
+
+
+class FileHashStore(HashStoreInterface):
```

Comment:
it feels like this should be in its own file (and possibly in a separate subdirectory alongside other implementations?), so that HashStore and HashStoreFactory can stay at a higher level of abstraction

Reply:
Done!